From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Mon, 5 Jul 2021 22:47:31 -0500
Subject: [PATCH] PlayerCandleCakeEvent

Sends the plugin the candle item used to create a candle block

diff --git a/src/main/java/com/empireminecraft/customevents/PlayerCandleCakeEvent.java b/src/main/java/com/empireminecraft/customevents/PlayerCandleCakeEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..5bfc7dc6f9b72bbbf96f096f39919ed1846b9972
--- /dev/null
+++ b/src/main/java/com/empireminecraft/customevents/PlayerCandleCakeEvent.java
@@ -0,0 +1,42 @@
+package com.empireminecraft.customevents;
+
+import org.bukkit.Location;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+public class PlayerCandleCakeEvent extends PlayerEvent {
+    private final Block cake;
+    private final ItemStack candle;
+
+    public PlayerCandleCakeEvent(@NotNull Player player, @NotNull Location cakeLoc, @NotNull ItemStack candle) {
+        super(player);
+        this.cake = cakeLoc.getBlock();
+        this.candle = candle;
+    }
+
+    @NotNull
+    public Block getCakeBlock() {
+        return cake;
+    }
+
+    @NotNull
+    public ItemStack getItem() {
+        return candle;
+    }
+
+    private static final HandlerList handlers = new HandlerList();
+
+    @NotNull
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
